### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by mreynolds0142**
Added user authentication to games controller for :new, :show, :create, :update.

### CODE REVIEW

**Comment by mreynolds0142** [comment]
build is failing because rubocop shows "app/controllers/games_controller.rb:1:1: W: Unnecessary disabling of Metrics/AbcSize."
Suggestions??  Can we remove the disable Metrics/AbcSize??

**Comment by jimxlin** [comment]
^ Yes, it means we can remove that commented line at the beginning of that file

File: spec/controllers/games_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe GamesController, type: :controller do
```
Comment (jimxlin) [inline spec/controllers/games_controller_spec.rb:3]: There should also be tests that verify that a logged in user can create, join, and view games.

Comment (mreynolds0142) [reply spec/controllers/games_controller_spec.rb:3]: Ok added logged in user tests

**Comment by jimxlin** [comment]
LGTM

**Pull request merged**